tensorflow/core/kernels/dml_kernel_wrapper.h

```diff
@@ -143,8 +143,8 @@ class DmlKernelWrapper : public DmlKernelWrapperBase {
     return std::make_shared<const typename TKernel::InitHelper>(ctx, attr_);
   }
 
-  DmlGpuEvent ComputeKernel(DmlKernel* kernel,
-                            DmlKernelContext* context) const override {
+  StatusOr<DmlGpuEvent> ComputeKernel(
```

Comment:
Does this ever actually return a Status?

Reply:
A few times. Before you would sometimes need to delegate the validation to an awkward void function if you wanted to validate something in Compute, just to be able to call OP_REQUIRES(...). Now, it's cleaner to simply call TF_RETURN_IF_ERROR and it avoids all the awkward `if (!ctx->status().ok()) return;` calls.